tensorflow_io/core/kernels/image_openexr_kernels.cc

```diff
@@ -0,0 +1,243 @@
+    }
+
+    Tensor* dtype_tensor = nullptr;
+    OP_REQUIRES_OK(context, context->allocate_output(1, TensorShape({static_cast<int64>(dtype.size()), maxchannel}), &dtype_tensor));
+    Tensor* channel_tensor = nullptr;
+    OP_REQUIRES_OK(context, context->allocate_output(2, TensorShape({static_cast<int64>(dtype.size()), maxchannel}), &channel_tensor));
+    for (size_t i = 0; i < dtype.size(); i++) {
+      for (size_t j = 0; j < dtype[i].size(); j++) {
+        dtype_tensor->flat<int64>()(i * maxchannel + j) = dtype[i][j];
+        channel_tensor->flat<string>()(i * maxchannel + j) = channel[i][j];
+      }
+      for (size_t j = dtype[i].size(); j < maxchannel; j++) {
+        dtype_tensor->flat<int64>()(i * maxchannel + j) = 0;
+        channel_tensor->flat<string>()(i * maxchannel + j) = "";
+      }
+    }
+  }
+ private:
+  mutex mu_;
+  Env* env_ GUARDED_BY(mu_);
+};
+class DecodeEXROp : public OpKernel {
+ public:
+  explicit DecodeEXROp(OpKernelConstruction* context)
+   : OpKernel(context) {
+     env_ = context->env();
+  }
+
+  void Compute(OpKernelContext* context) override {
+    const Tensor* input_tensor;
+    OP_REQUIRES_OK(context, context->input("input", &input_tensor));
+
+    const Tensor* index_tensor;
+    OP_REQUIRES_OK(context, context->input("index", &index_tensor));
+
+    const Tensor* channel_tensor;
+    OP_REQUIRES_OK(context, context->input("channel", &channel_tensor));
+
+    const string filename = "memory";
+    const string& memory = input_tensor->scalar<string>()();
+    std::unique_ptr<SizedRandomAccessFile> file(new SizedRandomAccessFile(env_, filename, memory.data(), memory.size()));
+    uint64 size;
+    OP_REQUIRES_OK(context, file->GetFileSize(&size));
+
+    OpenEXRIStream stream(filename, file.get(), size);
+    Imf::MultiPartInputFile input_file(stream);
+
+    const int64 index = index_tensor->scalar<int64>()();
+    const string& channel = channel_tensor->scalar<string>()();
+
+    Imf::InputPart input_part(input_file, index);
+
+    Imath::Box2i dw = input_part.header().dataWindow();
+
+    int64 height = dw.max.y - dw.min.y + 1;
+    int64 width  = dw.max.x - dw.min.x + 1;
+
+    DataType dtype = DT_INVALID;
+    Imf::PixelType pixel_type = Imf::UINT;
+    int x_sampling, y_sampling;
+    for (Imf::ChannelList::ConstIterator c = input_part.header().channels().begin(); c != input_part.header().channels().end(); c++) {
+      if (channel == c.name()) {
+        pixel_type = c.channel().type;
+        x_sampling = c.channel().xSampling;
+        y_sampling = c.channel().ySampling;
+        switch (pixel_type)
+        {
+        case Imf::UINT:
+          dtype = DT_UINT32;
+          break;
+        case Imf::HALF:
+          dtype = DT_HALF;
+          break;
+        case Imf::FLOAT:
+          dtype = DT_FLOAT;
+          break;
+        default:
+          OP_REQUIRES(context, false, errors::InvalidArgument("invalid data type: ", pixel_type));
```

Comment:
should this say "invalid pixel type"?

Reply:
Thanks. That makes sense. The PR has been updated.